modules/doc/doc_extract/doc_extractor.py

```diff
@@ -0,0 +1,118 @@
+from spire.doc import Document as SpireDocument, FileFormat
+from docx import Document as DocxDocument
+from pathlib import Path
+import dtlpy as dl
+import logging
+import shutil
+import nltk
+import os
+
+nltk.download('punkt')
+logger = logging.getLogger('pdf-to-text-logger')
+
+
+class DocExtractor(dl.BaseServiceRunner):
+
+    def doc_extraction(self, item: dl.Item, context: dl.Context) -> dl.Item:
+        """
+        Extracts a DOC/DOCX file item and uploads it as a TXT file.
+
+        Args:
+            context (dl.Context): Dataloop context to determine whether to extract tables from the original file.
+            item (dl.Item): Dataloop item, either DOC or DOCX file.
+
+        Returns:
+            dl.Item: New text item containing the extracted content in TXT format.
+        """
+        node = context.node
+        extract_tables = node.metadata['customNodeConfig']['extract_tables']
+        remote_path_for_extractions = node.metadata['customNodeConfig']['remote_path_for_extractions']
+        # Local test
+        # extract_tables = False
+        # remote_path_for_extractions = '/extracted_from_docs'
+
+        suffix = Path(item.name).suffix.lower()
+        if suffix not in {'.doc', '.docx'}:
+            raise ValueError("Only .doc and .docx files are supported for extraction.")
+
+        # Download path - original items
+        local_path = os.path.join(os.getcwd(), '../datasets', item.dataset.id, 'items',
+                                  os.path.dirname(item.filename[1:]))
+        os.makedirs(local_path, exist_ok=True)
+        item_local_path = item.download(local_path=local_path, save_locally=True)
+        # Convert .doc to .docx if necessary
+        if suffix == '.doc':
+            docx_path = os.path.join(local_path, f"{Path(item.name).stem}.docx")
+            try:
+                document = SpireDocument()
+                document.LoadFromFile(item_local_path)
+                document.SaveToFile(docx_path, FileFormat.Docx2016)
+                document.Close()
+
+            except Exception as e:
+                raise RuntimeError(f"Error converting item {item.id} to .docx format: {e}")
+
+        else:
+            docx_path = item_local_path
+
+        output_path = self.extract_content(docx_path=docx_path, local_path=local_path, extract_tables=extract_tables)
+
+        new_item = item.dataset.items.upload(local_path=output_path,
+                                             remote_path=remote_path_for_extractions,
+                                             item_metadata={
+                                                 'user': {'extracted_from_docs': True,
+                                                          'original_item_id': item.id}})
+
+        if new_item is None:
+            raise dl.PlatformException(f"No items was uploaded! local paths: {output_path}")
```

Comment:
Do we get the error for the upload if it failed? maybe we should print it

Reply:
the upload can return none